src/prefect/artifacts.py

```diff
@@ -173,6 +177,24 @@ async def format(self) -> str:
         return json.dumps(self._sanitize(self.table))
 
 
+class ProgressArtifact(Artifact):
+    progress: float
+    type: Optional[str] = "progress"
+
+    async def format(self) -> float:
```

Comment:
If we do decide to keep this method, it shouldn't be async

Reply:
@zangell44 I have it as async here to be compatible with the `Artifact` base class's `create` method, where it `await`s this method. Is there a better way to handle a pattern for async/sync classes in subclasses?